receiver/prometheusreceiver/internal/metricsbuilder.go

```diff
@@ -105,6 +108,16 @@ func (b *metricBuilder) AddDataPoint(ls labels.Labels, t int64, v float64) error
 		return fmt.Errorf("invalid sample: non-unique label names: %q", dupLabels)
 	}
 
+	defer func() {
+		if rerr == nil {
+			// Only mark this data point as in the current scrape
+			// iff it isn't a stale metric.
+			if !value.IsStaleNaN(v) {
```

Comment:
nit: you could combine these if-statements to reduce nesting.

Reply:
Done, thank you @dashpole!